integr8ly/integreatly-operator
# MGDAPI-3584 - e2e: A43 - update quota via ocm for addon installs

# Issue link

https://issues.redhat.com/browse/MGDAPI-3584

# What

* Update quota via OCM for addon installs for A34 e2e test to prevent flakiness of the test

# Verification steps
* Test log for master install - [integreatly-operator-test-k2gdb-test.log](https://github.com/integr8ly/integreatly-operator/files/11679486/integreatly-operator-test-k2gdb-test.log)
* Test log for addon install - [integreatly-operator-test-qflcp-test.log](https://github.com/integr8ly/integreatly-operator/files/11679493/integreatly-operator-test-qflcp-test.log)





If you would like to verify manually:

* Provision cluster
* Checkout this branch
* Install RHOAM
```
export LOCAL=false
make cluster/prepare/local
make code/run
```
* Allow RHOAM to complete install
* Run test and ensure test completes successfully
```
LOCAL=false TEST="A34" make test/e2e/single
```
* Delete RHMI CR and uninstall RHOAM

Once RHOAM completes uninstall:
* Install RHOAM addon
* Update RHMI CR spec to `useClusterStorage: true` to progress installation
* Allow RHOAM addon to complete
* Ensure the ocm token and cluster id of the cluster is set and run the test
```
export OCM_TOKEN=<OCM_TOKEN>
export CLUSTER_ID=$(ocm get clusters --parameter search="name like '<cluster_name>'" | jq -r '.items[].id')
LOCAL=false TEST="A34" make test/e2e/single
```
  * During the test you can verify that the ocm quota parameter is updated using another terminal
  ```
  ocm get /api/clusters_mgmt/v1/clusters/$CLUSTER_ID/addons/managed-api-service
  ```
  * Verify test passes successfully
